controllers/articlesController.js

```diff
@@ -269,4 +250,177 @@ export default class PostController {
+            as: "likes",
+            attributes: ["username"],
+            through: { attributes: [] }
+          },
+          {
+            model: Tag,
+            attributes: ["name"],
+            as: "tagsList",
+            through: { attributes: [] }
+          }
+        ]
+      });
+      if (all.length) {
+        const articles = all
+          .map(each => {
+            const {
+              author: { followers, username, bio, image },
+              tagsList,
+              likes,
+              ...article
+            } = each.get();
+            const following = !!_.dropWhile(followers, f => f.id !== id).length;
+            return {
+              ...article,
+              likes: likes.map(like => like.username),
+              author: { following, username, bio, image },
+              favorited: !!likes.length,
+              favoritesCount: likes.length,
+              tagsList: tagsList.map(tag => tag.get().name || null)
+            };
+          })
+          .filter(art => liked === null || art.likes.includes(liked))
+          .filter(art => oneTag === null || art.tagsList.includes(oneTag));
+        return res.status(OK).json({
+          message: "Successful",
+          articles,
+          articlesCount: articles.length
+        });
+      }
+      return res.status(NO_CONTENT).json({ message: "No articles created yet!" });
+    } catch (error) {
+      return res
+        .status(INTERNAL_SERVER_ERROR)
+        .json({ message: "Can't get the articles, server error" });
+    }
+  }
+
+  /**
+   * @description It helps the user to update any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async update(req, res) {
+    try {
+      const {
+        params: { slug },
+        user: { id: userId },
+        body
+      } = req;
+      const article = await Article.findOne({ where: { slug, userId } });
+      if (article) {
+        const allowed = _.pick(body, Object.keys(articleSchema));
+        const valid = await articleValidator(
+          allowed,
+          _.pick(articleSchema, Object.keys(allowed))
+        );
+        const { tagsList = [], ...rest } = valid;
+        if (tagsList.length) {
+          tagsList.map(async t => {
+            const tag = await Tag.findOrCreate({ where: { name: t } }).spread(
+              tg => tg
+            );
+            await article.addTagsList(tag);
+          });
+        }
+        const updated = await article.update(rest, {
+          where: { id: article.id, userId },
+          returning: true,
+          limit: 1
+        });
+        return res.status(CREATED).json({
+          article: {
+            ...updated.get(),
+            author: _.pick(await updated.getAuthor(), ["username", "bio", "image"]),
+            tagsList: await updated.getTagsList({ attributes: ["name"] }).map(t => {
```

Comment:
@Musigwa You don't need the bracket for map, you can just return the statement directly after the arrow

Reply:
Thanks for the reminder, I have applied the feedback given.